common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/connection/ConnectionFactoryImpl.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.connection;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Objects;
+import java.util.Properties;
+
+/**
+ * A ConnectionFactory implementation that utilizes the ConnectionFactoryConfig for connection creation.
+ */
+public class ConnectionFactoryImpl implements ConnectionFactory {
+    private final String jdbcUrl;
+    private final Properties properties;
+    private final boolean autoCommit;
+
+    public ConnectionFactoryImpl(@NotNull ConnectionFactoryConfig connectionFactoryConfig) {
+        Objects.requireNonNull(connectionFactoryConfig, "connectionFactoryConfig");
+
+        this.jdbcUrl = Objects.requireNonNull(connectionFactoryConfig.getUri()).toString();
+        this.properties = getProperties(connectionFactoryConfig);
+        this.autoCommit = connectionFactoryConfig.getAutoCommit();
+    }
+
+    private Properties getProperties(ConnectionFactoryConfig connectionFactoryConfig) {
+        Properties properties = new Properties();
+        addProperty(properties, "user", connectionFactoryConfig.getUserName());
```

Comment:
Are these random string values or is there a rule/expectation from another module? Are the values maybe specified by another library (java.sql)?

Reply:
Those property keys are coming from the [postgres jdbc documentation](https://jdbc.postgresql.org/documentation/head/connect.html) - will note that down as javadoc.